core/src/main/scala/org/apache/spark/sql/execution/columnar/ExternalStoreUtils.scala

```diff
@@ -493,7 +493,7 @@ object ExternalStoreUtils {
     }
   }
 
-  private def removeCachedObjects(table: String): () => Iterator[Unit] = () => {
+  def removeCachedObjects(table: String): () => Iterator[Unit] = () => {
     ConnectionPool.removePoolReference(table)
     CodeGeneration.removeCache(table)
```

Comment:
For this cleanup may be we can get a handle of the generated code for batch insert for a particular table which is dropped later.

So create a table.
Do df insert.
Check whether code got generated for batch insert.
If yes get the execution key.
Drop the table.
Check in the generated code cache for this key. It should be clean.